### DISCUSSION TITLE
@walletconnect/modal-react-native does not support connection to multiple chains

### DISCUSSION THREAD

**Discussion by YerimMoral3s** [WalletConnect/walletconnect-monorepo#3226] [Developer Support]
when the component is created and we pass the session params like this: 
 `const sessionParams = {
  namespaces: {
    eip155: {
      methods: compatibleSafeMethods,
      chains: ['eip155:1', 'eip155:137'],
      events: ['chainChanged', 'accountsChanged'],
      rpcMap: {}
    }
  }
}` 
to WalletConnectModal, so that everything works correctly.

but the problem is visible when you try to connect to another application, for example metamask, but the error is persistent with other wallets. When you try to connect, the app just opens and nothing connects.

but if you change session params to:
`const sessionParams = {
  namespaces: {
    eip155: {
      methods: compatibleSafeMethods,
      chains: ['eip155:1'],
      events: ['chainChanged', 'accountsChanged'],
      rpcMap: {}
    }
  }
}`
and repeat the process, the connection is established correctly

**Comment by glitch-txs** [comment]
So far this issue is only on MetaMask side. However you can still add an optionalNamespaces object with the rest of the chains you want to connect to, the wallet should connect whether it supports those chains or not.
